logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -593,6 +583,71 @@ private UptimeMetric initOrGetUptimeMetric(final ThreadContext context,
         return retrievedMetric.toJava(UptimeMetric.class);
     }
 
+    private void initializePqFlowMetrics(final ThreadContext context, final RubySymbol[] flowNamespace, final Metric<Number> uptimeInPreciseSeconds) {
```

Comment:
We should pass our actual `UptimeMetric` through to avoid mixing scales accidentally (requires change to caller):

```suggestion
    private void initializePqFlowMetrics(final ThreadContext context, final RubySymbol[] flowNamespace, final UptimeMetric uptime) {
        final Metric<Number> uptimeInPreciseSeconds = uptimeMetric.withUnitsPrecise(SECONDS);
```